sys/kern/ringbuf.c

```diff
@@ -10,29 +10,31 @@ void ringbuf_init(ringbuf_t *rb, void *buf, size_t size) {
   rb->data = buf;
 }
 
-static void produce(ringbuf_t *buf, unsigned bytes) {
-  assert(buf->count + bytes <= buf->size);
-  assert(buf->head + bytes <= buf->size);
+/* below two functions are to help to track ringbuf pointers in case when it is
+ * changed by something else. e.q. DMA */
+void ringbuf_produce(ringbuf_t *buf, unsigned bytes) {
+  assert(bytes <= buf->size - buf->count);
   buf->count += bytes;
   buf->head += bytes;
-  if (buf->head == buf->size)
-    buf->head = 0;
+  if (buf->head >= buf->size)
+    buf->head -= buf->size;
+  assert(buf->head < buf->size);
 }
 
-static void consume(ringbuf_t *buf, unsigned bytes) {
+void ringbuf_consume(ringbuf_t *buf, unsigned bytes) {
   assert(buf->count >= bytes);
-  assert(buf->tail + bytes <= buf->size);
   buf->count -= bytes;
   buf->tail += bytes;
-  if (buf->tail == buf->size)
-    buf->tail = 0;
+  if (buf->tail >= buf->size)
+    buf->tail -= buf->size;
+  assert(buf->tail < buf->size);
 }
 
 bool ringbuf_putb(ringbuf_t *buf, uint8_t byte) {
```

Comment:
You should describe what that function does. Like with two above.

Reply:
Sorry, I am not an author of this function but imo this name suits well to what this function does (_putb_ -> put byte)